src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -110,6 +123,20 @@ private static double GetFragmentation()
         }
 #endif
 
+#if NET6_0_OR_GREATER
+        private static IEnumerable<Measurement<long>> GetGarbageCollectionHeapSizes()
+        {
+            var generationInfo = GC.GetGCMemoryInfo().GenerationInfo;
```

Comment:
we need some assert that the length is no more than 5, else HeapNames will outofboundexception

Reply:
I thought to use `i < measurements.Length && i < HeapNames.Length` as for loop termination check, or use `i < HeapName.Length ? HeapNames[i] : $"Gen{i}"` as generation name, but it feels like waste of checks **if** we can assume the garbage collection algorithm does not change.

Do you have any suggestion? Or a simple `Assert(generationInfo.Length <= HeapNames.Length, "There should not be more than 5 generations");` will do?

Reply:
I've added an additional check to work if the size goes beyond 5.

Reply:
> the garbage collection algorithm does not change.

^ Can you leverage this, and build readonly array in the ctor?

Reply:
Done. I've moved them to a readonly array.